glog.go

```diff
@@ -38,136 +98,167 @@ func Flush() {
 	zap.L().Sync()
 }
 
-// V is a shim
+// skipLogger constructs a suggared logger with an additional caller skip to
+// ensure we log the correct line number.
+func skipLogger() *zap.SugaredLogger {
+	return zap.L().WithOptions(zap.AddCallerSkip(1)).Sugar()
```

Comment:
Could this be allocated and reused? As is, this could cause a lot of garbage.

Reply:
It could be allocated in `func int()` but callers would need to swap in their own configured logger explicitly via an analogue of [`zap.ReplaceGlobals()`](https://godoc.org/go.uber.org/zap#ReplaceGlobals) which would mean the package was no longer a complete drop-in replacement for glog.

With this implementation, the user can just call [`zap.ReplaceGlobals()`](https://godoc.org/go.uber.org/zap#ReplaceGlobals) directly and we'll use their configured logger.

On reflection maybe the performance improvement is worth the tradeoff?